mempool/reactor.go

```diff
@@ -21,20 +22,35 @@ const (
 	maxTxSize  = maxMsgSize - 8 // account for amino overhead of TxMessage
 
 	peerCatchupSleepIntervalMS = 100 // If peer is behind, sleep this amount
+  
+  // UnknownPeerID is the peer ID to use when running CheckTx when there is
+	// no peer (e.g. RPC)
+	UnknownPeerID uint16 = 0
 )
 
 // MempoolReactor handles mempool tx broadcasting amongst peers.
+// It maintains a map from peer ID to counter, to prevent gossiping txs to the
+// peers you received it from.
 type MempoolReactor struct {
 	p2p.BaseReactor
 	config  *cfg.MempoolConfig
 	Mempool *Mempool
+
+	idMtx     sync.RWMutex
+	nextID    uint16 // assumes that a node will never have over 65536 active peers
+	peerMap   map[p2p.ID]uint16
```

Comment:
What is size of p2p.ID, i.e., what is the gain of this indirection?

Reply:
It's 20 bytes. The gain is to store max 2 bytes with each tx to identify the sender, rather than 20